test/test_connectionpool.py

```diff
@@ -308,7 +311,91 @@ def kaboom(*args, **kwargs):
 
         new_pool_size = c.pool.qsize()
         self.assertEqual(initial_pool_size, new_pool_size)
-
+    
+    def test_preserves_request_header_order(self):
+        """
+        Ensure that headers provided to ConnectionPool.urlopen()
+        are sent to the underlying httplib in the same order.
+        """
+        class AbortRequest(BaseException):
+            pass
+        
+        expected_request_headers = OrderedDict([('X-Header-%d' % i, str(i)) for i in range(16)])
```

Comment:
Nothing too special. Just a large enough number that it's a low probability that (shuffle(X) == X), where X is a list of headers. Maybe I'll change it to something like 8 just so that it's not a power of 2 (which might imply the number was special). :-)